src_c/surface.c

```diff
@@ -499,8 +499,23 @@ surface_str(PyObject *self)
         return PyUnicode_FromString("<Surface(Dead Display)>");
     }
 
-    return PyUnicode_FromFormat("<Surface(%dx%dx%d SW)>", surf->w, surf->h,
-                                surf->format->BitsPerPixel);
+    PyObject *colorkey = surf_get_colorkey((pgSurfaceObject *)self, NULL);
+    PyObject *global_alpha = surf_get_alpha((pgSurfaceObject *)self, NULL);
+
+    // 46 is max len of format str, plus null terminator
+    char format[47] = "<Surface %dx%dx%d";
+    if (PyObject_IsTrue(colorkey))
+        strcat(format, " colorkey=%S");
+    if (PyObject_IsTrue(global_alpha))
+        strcat(format, " global_alpha=%S");
```

Comment:
I think there should be commas in between the different "attributes" so like:
```suggestion
        strcat(format, ", colorkey=%S");
    if (PyObject_IsTrue(global_alpha))
        strcat(format, ", global_alpha=%S");
```

Reply:
Sure, I was mainly following from the issue